module/event-sourcing/src/Application/DependencyInjection/CompilerPass/ProjectorCompilerPass.php

```diff
@@ -61,9 +61,12 @@ private function processServices(ContainerBuilder $container): void
             }
 
             $parameters = $method->getParameters();
-            $eventClass = (string) $parameters[0]->getType();
 
-            $definition->addMethodCall('add', [new Reference($serviceId), $eventClass]);
+            $class = $parameters[0]->getClass();
+            if ($class) {
+                $eventClass = $class->getName();
+                $definition->addMethodCall('add', [new Reference($serviceId), $eventClass]);
+            }
```

Comment:
if does not have a class I guess the exception should be thrown because we cannot recognize the type based on it?